### PR TITLE
Remove HOTP support

### PR DESCRIPTION
**Pull request by codingllama**
While looking up github.com/gokyle/hotp I found some old deprecation warnings and decided to address them.

### CODE REVIEW

**Comment by codingllama** [comment]
@russjones @r0mant I don't have a lot of context on this one, apart from that it looks safe to delete. Could you tell whether we should be keeping this?

**Review by zmb3 — APPROVED** [review]

File: lib/auth/password.go [hunk 1]
```diff
@@ -201,64 +201,35 @@ func (s *Server) checkPassword(user string, password []byte, otpToken string) (*
 // checkOTP determines the type of OTP token used (for legacy HOTP support), fetches the
```
Comment (espadolini) [inline lib/auth/password.go:201]:
```suggestion
// checkOTP fetches the
```

**Review by espadolini — APPROVED** [review]

File: constants.go [hunk 2]
```diff
@@ -389,8 +389,6 @@ type OTPType string
 const (
 	// TOTP means Time-based One-time Password Algorithm (for Two-Factor Authentication)
 	TOTP = OTPType("totp")
```
Comment (espadolini) [inline constants.go:391]: Do we even need `OTPType` anymore?

File: lib/auth/password.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/password.go:201]: Done.

File: constants.go (hunk 2 above)
Comment (codingllama) [reply constants.go:391]: Good call, removed.

**Comment by codingllama** [comment]
Addressed @espadolini's comments and removed a few more HOTP references, PTAL.

There's one reference in docs, I'll do a separate PR for it.

**Review by espadolini — APPROVED** [review]

**Comment by espadolini** [comment]
Thank you for being thorough, as usual.

**Comment by codingllama** [comment]
https://github.com/gravitational/teleport/pull/16590

**Pull request merged**